From 5e4563db30c169cfe8cc34b5c83f5b88e845f2f8 Mon Sep 17 00:00:00 2001
From: Matthias Bach <marix@marix.org>
Date: Thu, 5 Feb 2015 00:04:49 +0100
Subject: [PATCH] Fix interaction of the updater with headless gridcoin
 research

The updater assumed the binary name to be gridcoind instead of
gridcoinresearchd.
---
 CompilingGridcoinOnLinux.txt | 4 ++--
 src/upgrader.cpp             | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/CompilingGridcoinOnLinux.txt b/CompilingGridcoinOnLinux.txt
index d7917ed..c418348 100644
--- a/CompilingGridcoinOnLinux.txt
+++ b/CompilingGridcoinOnLinux.txt
@@ -50,7 +50,7 @@ chmod 755 leveldb/build_detect_platform
 make -f makefile.unix USE_UPNP=- 
 
 
-##### End of Step 1: gridcoind will be found in /src directory
+##### End of Step 1: gridcoinresearchd will be found in /src directory
 
 
 
@@ -81,7 +81,7 @@ addnode=node.gridcoin.us
 
 #save and exit.  Leave the lines with ssl in them out if you don't need ssl.
 
-####### Testing Gridcoin gridcoind :
+####### Testing Gridcoin gridcoinresearchd :
 #Let server start
 gridcoinresearchd
 #Should return current difficulty:
diff --git a/src/upgrader.cpp b/src/upgrader.cpp
index 092874e..4d0927e 100644
--- a/src/upgrader.cpp
+++ b/src/upgrader.cpp
@@ -167,7 +167,7 @@ int main(int argc, char *argv[])
             }
         }
     }
-    else if (strcmp(argv[1], "gridcoind")==0)
+    else if (strcmp(argv[1], "gridcoinresearchd")==0)
     {
         if (waitForParent(atoi(argv[2])))
         {
@@ -634,7 +634,7 @@ std::string Upgrader::targetswitch(int target)
         return "gridcoinresearch";
 
         case DAEMON:
-        return "gridcoind";
+        return "gridcoinresearchd";
 
         case UPGRADER:
         return "gridcoinupgrader";
@@ -658,4 +658,4 @@ Upgrader::~Upgrader()
     unlockTarget();
     downloadThread.interrupt();
     downloadThread.join();
-}
\ No newline at end of file
+}
